### DISCUSSION TITLE
[FR] Support kimi.ai

### DISCUSSION THREAD

**Discussion by wwjCMP** [logancyang/obsidian-copilot#541] [Q&A]
https://platform.moonshot.cn/console/api-keys

Another powerful online online LLM

**Comment by wwjCMP** [comment]
https://platform.moonshot.cn/docs/intro#%E4%B8%BB%E8%A6%81%E6%A6%82%E5%BF%B5

**Comment by wwjCMP** [comment]
It is compatible with the OpenAI API format. I think it would be better to provide an option for these OpenAI-compatible APIs. This way, there's no need for individual adaptation.

Provide an API input box, model name, and offer endpoint box.

It can refer to the approach of text generator.

**Comment by logancyang** [comment]
Have you tried the endpoint override? Does it fail with CORS as well?

**Comment by wwjCMP** [comment]
> Have you tried the endpoint override? Does it fail with CORS as well?

No, it can be used normally in the text generator. Perhaps it's due to network issues?

**Comment by BaoLiqi** [comment]
> > Have you tried the endpoint override? Does it fail with CORS as well?
> 
> No, it can be used normally in the text generator. Perhaps it's due to network issues?

I had a CORS problem until I set up a proxy server.

> Access to fetch at 'https://api.moonshot.cn/v1/chat/completions' from origin 'app://obsidian.md' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

**Comment by wwjCMP** [comment]
Additionally, it is great to allow users to save multiple custom models.

**Comment by logancyang** [comment]
Any openai replacement should work without CORS error with the new local proxy setting from https://github.com/logancyang/obsidian-copilot/pull/495

**Comment by wwjCMP** [comment]
> Any openai replacement should work without CORS error with the new local proxy setting from #495

This can indeed be used, but custom model names cannot be used.

**Comment by logancyang** [comment]
@wwjCMP this field doesn't work for you? When you fill this field, you must pick any *OpenAI* model in the dropdown as a placeholder.
<img width="858" alt="SCR-20240814-taqm" src="https://github.com/user-attachments/assets/a18b5f83-7176-482d-abd2-a9f13ec5cc72">

If it still doesn't work, please open an issue and show how it fails, I can take a look.

**Comment by wwjCMP** [comment]
> @wwjCMP this field doesn't work for you? When you fill this field, you must pick any _OpenAI_ model in the dropdown as a placeholder. <img alt="SCR-20240814-taqm" width="858" src="https://private-user-images.githubusercontent.com/4860545/358126476-a18b5f83-7176-482d-abd2-a9f13ec5cc72.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3MDI1MTUsIm5iZiI6MTcyMzcwMjIxNSwicGF0aCI6Ii80ODYwNTQ1LzM1ODEyNjQ3Ni1hMThiNWY4My03MTc2LTQ4MmQtYWJkMi1hOWYxM2VjNWNjNzIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTVUMDYxMDE1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzZlMmFmNTA5ODEyNTZkNTU5ZDI3Y2M5MGJjN2U0MTVmOTZhZjUyZTM0Zjg3MTg4YmRiZjlkOTY4ZDQ4OTZkMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.gcDhFH7rOEuSWlwuYXxTM1d5C2WErfsessQ2C6uLgGo">
> 
> If it still doesn't work, please open an issue and show how it fails, I can take a look.

In fact, I can make it work by inputting the name of an OpenAI model, such as GPT-4. This way, I can use a reverse proxy to utilize third-party services compatible with OpenAI. However, I cannot specify the model provided by the third-party service, such as specifying the model as kimi-v2.

**Comment by wwjCMP** [comment]
> @wwjCMP this field doesn't work for you? When you fill this field, you must pick any _OpenAI_ model in the dropdown as a placeholder. <img alt="SCR-20240814-taqm" width="858" src="https://private-user-images.githubusercontent.com/4860545/358126476-a18b5f83-7176-482d-abd2-a9f13ec5cc72.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3MzE5OTMsIm5iZiI6MTcyMzczMTY5MywicGF0aCI6Ii80ODYwNTQ1LzM1ODEyNjQ3Ni1hMThiNWY4My03MTc2LTQ4MmQtYWJkMi1hOWYxM2VjNWNjNzIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTVUMTQyMTMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODhhZDEyYzM3YTZiM2NmNWU4ZTgyMTE3NDdhMDU5MzE5MDJjZDdjYjQ4ZWY4MDBlYWQ4NDk4ZjRlZDU2YWJiMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.IxQ08oYlfetec3BMzT5EWtLOtaHETb5idQp1GLm8l7A">
> 
> If it still doesn't work, please open an issue and show how it fails, I can take a look.
![image](https://github.com/user-attachments/assets/c2a7865b-1f62-419c-8a99-79f224c3abb4)

Ok, now i can use the custom model name by change this at the same time. But now only one  third-party model can be added.
